qemu/tests/qemu_guest_agent.py

```diff
@@ -454,12 +454,44 @@ def gagent_check_set_user_password(self, test, params, env):
     @error_context.context_aware
     def gagent_check_get_vcpus(self, test, params, env):
         """
-        Execute "guest-set-vcpus" command to guest agent
+        Execute "guest-get-vcpus" command to guest agent.
+
+        Steps:
+        1) Check cpu info from guest agent.
+        2) Check cpu number.
+
         :param test: kvm test object
         :param params: Dictionary with the test parameters
         :param env: Dictionary with test environment.
         """
-        self.gagent.get_vcpus()
+        error_context.context("Check cpu info from guest agent.", logging.info)
+        vcpus_info = self.gagent.get_vcpus()
+        cpu_num_qga = len(vcpus_info)
+        for vcpu in vcpus_info:
+            if params.get("os_type") == "linux" and not vcpu["can-offline"]:
+                test.fail("Linux guest cpu can't be offline from qga"
+                          "which isn't expected.")
+            if self.params.get("os_type") == "windows" and vcpu["can-offline"]:
```

Comment:
Please unify the way of getting os_type, Line 471 and Line 474

Reply:
updated.